### PR TITLE
Add batch action support

### PR DESCRIPTION
**Pull request by cjyclaire**
Adding `batchAction` support.

Note: Files in `/gems` are generated by `rake build:svc`, added in this PR as reference, will be removed before merged

cc:\ @trevorrowe

### CODE REVIEW

**Comment by cjyclaire** [comment]
This PR has been updated and ready for review.

File: build_tools/aws-sdk-code-generator/spec/aws-sdk-code-generator/generators/resource/collection_class_spec.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          'batchActions' => {
+             'Terminate' => {
+              'request' => {
+                'operation' => 'TerminateConcerts',
+                'params' => [
+                  { 'target' => 'Concerts[*].ConertName',
+                    'source' => 'identifier',
+                    'name' => 'ConcertName',
+                  },
+                  { 'target' => 'Concerts[*].ConertLocation',
```
Comment (awood45) [inline build_tools/aws-sdk-code-generator/spec/aws-sdk-code-generator/generators/resource/collection_class_spec.rb]: Same typo question for `ConertLocation`

File: build_tools/aws-sdk-code-generator/spec/aws-sdk-code-generator/generators/resource/collection_class_spec.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          }
+        }}
+
+        let(:terminate_sample) {{
+          'batchActions' => {
+             'Terminate' => {
+              'request' => {
+                'operation' => 'TerminateConcerts',
+                'params' => [
+                  { 'target' => 'Concerts[*].ConertName',
```
Comment (cjyclaire) [inline build_tools/aws-sdk-code-generator/spec/aws-sdk-code-generator/generators/resource/collection_class_spec.rb]: Typo indeed, thanks for catching

**Comment by trevorrowe** [comment]
Great work! I've merged this into the cod-generation branch.

**Pull request merged**